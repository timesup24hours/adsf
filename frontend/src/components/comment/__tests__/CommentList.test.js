import CommentListWrapper, { CommentList } from '../CommentList'
import getMuiTheme from 'material-ui/styles/getMuiTheme';

const comments = {

}
const commentId = 'comentId'
const comment = 'testComment'
const editedComment = 'editedComment'
const snackbar = {
  open: true,
  message: '13'
}
const auth = {
  user: {
    _id: '123',
    local: {
      nickname: '123',
      username: '123'
    }
  }
}
// const d = new Date();
const d = '2014-04-03T01:40:41.448Z'
const commentDeleteRequest = (id) => expect(id).toBe(commentId)
const commentEditRequest = ({ id, c }) => expect({ id, c }).toBe({ id: commentId, comment })
const snackbarClose = () => jest.fn()

const store = mockStore({ comments, snackbar, auth })
const componentWrapper = (<CommentListWrapper
  commentId={commentId}
  key='123'
  user={auth.user}
  loginUserId='123'
  createdAt={d}
  comment={comment}
  store={store}
/>)

const component = (<CommentList
  commentId={commentId}
  key='123'
  user={auth.user}
  loginUserId='123'
  createdAt={d}
  comment={comment}
  snackbar={snackbar}
  auth={auth}
  comments={comments}
/>)

const componentWithDispatch = (<CommentList
  commentId='123'
  key='123'
  user={auth.user}
  loginUserId='123'
  createdAt={d}
  comment={comment}
  snackbar={snackbar}
  auth={auth}
  comments={comments}
  commentDeleteRequest={commentDeleteRequest}
  commentEditRequest={commentEditRequest}
  snackbarClose={snackbarClose}
/>)

test('# CommentListWrapper', () => {
  const wrapper = shallow(componentWrapper)
  expect(wrapper).toMatchSnapshot()
})

test('# CommentList itself', () => {
  const shallowCom = shallow(component)
  expect(shallowCom).toMatchSnapshot()
})

test('# CommentList with dispatch', () => {
  const shallowCom = shallow(componentWithDispatch)
  expect(shallowCom).toMatchSnapshot()
})

test('# CommentList call componentDidMount', () => {
  sinon.spy(CommentList.prototype, 'componentDidMount');
  const app = mount(componentWithDispatch, {
      context: {
        muiTheme: getMuiTheme(),
      },
      childContextTypes: {
        muiTheme: React.PropTypes.object.isRequired,
      },
    });

  expect(CommentList.prototype.componentDidMount.calledOnce).toEqual(true);
  expect(app.state().comment).toBe(comment)
})

test('# CommentList edit comment', () => {
  const app = mount(componentWithDispatch, {
      context: {
        muiTheme: getMuiTheme(),
      },
      childContextTypes: {
        muiTheme: React.PropTypes.object.isRequired,
      },
    });

  expect(app.state().comment).toBe(comment)
  app.find('#CommentList-edit').simulate('click')
})
