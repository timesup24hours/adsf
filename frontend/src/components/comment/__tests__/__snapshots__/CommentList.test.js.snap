exports[`test # CommentList itself 1`] = `
<div
  className="CommentList-container">
  <div
    className="CommentList-user-holder">
    <div
      className="CommentList-user-group">
      <div
        className="CommentList-avatar">
        <UserAvatar
          user={
            Object {
              "_id": "123",
              "local": Object {
                "nickname": "123",
                "username": "123",
              },
            }
          } />
      </div>
      <div
        className="CommentList-username-time-group">
        <div
          className="CommentList-username">
          123
        </div>
        <div
          className="CommentList-time">
          3 years ago
        </div>
      </div>
    </div>
  </div>
  <div
    className="CommentList-comment">
    <div
      className="CommentList-comment-show" />
  </div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false} />,
        <FlatButton
          disabled={false}
          keyboardFocused={true}
          label="Delete"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false} />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
    title="Delete Confirm">
    This comment would be deleted permanently.
  </Dialog>
  <div
    className="CommentList-edit"
    id="CommentList-edit"
    onTouchTap={[Function]}>
    <div>
      <EditorBorderColor />
    </div>
  </div>
  <div
    className="CommentList-delete"
    onTouchTap={[Function]}>
    <div>
      <ActionDeleteForever />
    </div>
  </div>
  <Snackbar
    autoHideDuration={5000}
    message="13"
    onRequestClose={[Function]}
    open={true} />
</div>
`;

exports[`test # CommentList with dispatch 1`] = `
<div
  className="CommentList-container">
  <div
    className="CommentList-user-holder">
    <div
      className="CommentList-user-group">
      <div
        className="CommentList-avatar">
        <UserAvatar
          user={
            Object {
              "_id": "123",
              "local": Object {
                "nickname": "123",
                "username": "123",
              },
            }
          } />
      </div>
      <div
        className="CommentList-username-time-group">
        <div
          className="CommentList-username">
          123
        </div>
        <div
          className="CommentList-time">
          3 years ago
        </div>
      </div>
    </div>
  </div>
  <div
    className="CommentList-comment">
    <div
      className="CommentList-comment-show" />
  </div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false} />,
        <FlatButton
          disabled={false}
          keyboardFocused={true}
          label="Delete"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false} />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
    title="Delete Confirm">
    This comment would be deleted permanently.
  </Dialog>
  <div
    className="CommentList-edit"
    id="CommentList-edit"
    onTouchTap={[Function]}>
    <div>
      <EditorBorderColor />
    </div>
  </div>
  <div
    className="CommentList-delete"
    onTouchTap={[Function]}>
    <div>
      <ActionDeleteForever />
    </div>
  </div>
  <Snackbar
    autoHideDuration={5000}
    message="13"
    onRequestClose={[Function]}
    open={true} />
</div>
`;

exports[`test # CommentListWrapper 1`] = `
<CommentList
  auth={
    Object {
      "user": Object {
        "_id": "123",
        "local": Object {
          "nickname": "123",
          "username": "123",
        },
      },
    }
  }
  comment="testComment"
  commentDeleteRequest={[Function]}
  commentEditRequest={[Function]}
  commentId="comentId"
  createdAt="2014-04-03T01:40:41.448Z"
  loginUserId="123"
  snackbar={
    Object {
      "message": "13",
      "open": true,
    }
  }
  snackbarClose={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  user={
    Object {
      "_id": "123",
      "local": Object {
        "nickname": "123",
        "username": "123",
      },
    }
  } />
`;
